### DISCUSSION TITLE
All workflow_dispatch to only run from a specific branch set at the repo level

### DISCUSSION THREAD

**Discussion by scott-doyland-burrows** [community/community#19779] [Actions]
Hi,

I would like to set ```workflow_dispatch``` within a workflow - but at the repo level only allow the ```main``` branch to trigger the workflow manually.  Here is my use-case:

1. I have a workflow that runs on a ```schedule``` and on a ```workflow_dispatch```.
2. I cannot use environment protection to request approvals for the ```workflow_dispatch```, as this would then also request approvals for the ```schedule```.
3. I do not want an actor to be able to create a ```non-main``` branch and run the ```workflow_dispatch``` from that branch, as they could then update the workflow to do something undesirable.
4. If the ```workflow_dispatch``` could only run from the ``` main``` branch then I could ensure the workflow was always correct.
5. There should be a repo wide setting that allows me to select which branches can run the ```workflow_dispatch```, and this should not be able to be overridden by setting anything in the workflow itself.

Thanks.

**Comment by MatthewCane** [comment]
Hi @scott-doyland-burrows, did you find a solution to this? I am facing a similar issue.

**Reply by scott-doyland-burrows** [reply]
No. It's one of the big flaws in GHAs. 

That, and the fact that anyone with write access to a repo can obtain GHAs secrets. 

Hopefully GitHub will eventually redesign some of this.

**Comment by bradsk88** [comment]
Minor, but should the title of this discussion be updated to say "Allow" instead of "All"?

Might make it more likely to be spotted.

**Comment by mohamedmansour33** [comment]
تراخيص الاستخدام وضبط المعرفات الاساسيه

**Comment by smariko** [comment]
+1

**Comment by dongvt** [comment]
A workaround for me was just to put a job as parent of all other jobs and make it run just when `ref == 'main'`
